flask_datadog.py

```diff
@@ -98,6 +111,85 @@ def decr(self, *args, **kwargs):
         """Helper to expose `self.statsd.decrement` under a shorter name"""
         return self.statsd.decrement(*args, **kwargs)
 
+    def setup_middleware(self):
+        """Helper to configure/setup any Flask-Datadog middleware"""
+        # Configure response time middleware (if desired)
+        self.config.setdefault('DATADOG_CONFIGURE_MIDDLEWARE', True)
+        self.config.setdefault('DATADOG_RESPONSE_METRIC_NAME', 'flask.response.time')
+        self.config.setdefault('DATADOG_RESPONSE_SAMPLE_RATE', 1)
+        self.config.setdefault('DATADOG_RESPONSE_AUTO_TAG', True)
+        if self.config['DATADOG_CONFIGURE_MIDDLEWARE']:
+            self.app.before_request(self.before_request)
+            self.app.after_request(self.after_request)
+
+    def before_request(self):
+        """
+        Flask-Datadog middleware handle for before each request
+        """
+        # Set the request start time
+        g.flask_datadog_start_time = time.time()
+        g.flask_datadog_request_tags = []
+
+        # Add some default request tags
+        if self.config['DATADOG_RESPONSE_AUTO_TAG']:
+            self.add_request_tags(['endpoint:%s' % (str(request.endpoint).lower(), ),
```

Comment:
Sure, not a problem, assume we should do the same for `method` as well, if we let one be configurable, might as well make the other as well?

/me knows the pain of trying to reconfigure a bunch of dashboards